### PR TITLE
Maps projects on home page

### PR DESCRIPTION
**Pull request by dereklieu**
This includes #74. @kamicut can you give it a code review? Anna is working on fixing styles for this, but I think we can go ahead and sync it up with master, and then merge her work in separately.

### CODE REVIEW

File: app/assets/scripts/components/map.js [hunk 1]
```diff
@@ -1,30 +1,83 @@
...
+
+  addClusterMarkers: function (markers) {
+    const markerLayer = this.markerLayer;
+    markerLayer.clearLayers();
+    markers.forEach(function (marker) {
+      const leafletMarker = L.marker(marker.centroid, {
+        icon: L.mapbox.marker.icon({'marker-symbol': 'garden', 'marker-color': '0044FF'})
+      });
+      leafletMarker.bindPopup(
+        `<div class='marker__internal'>` +
```
Comment (kamicut) [inline app/assets/scripts/components/map.js]: @derek for future reference, template strings can be multiline. No need to use a plus sign https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: app/assets/scripts/components/map.js [hunk 2]
```diff
@@ -1,30 +1,83 @@
...
+      markerLayer.addLayer(leafletMarker);
+    });
   },
 
   componentWillUnmount: function () {
     this.map.remove();
   },
 
+  componentWillReceiveProps: function (newProps) {
+    if (newProps.location && (!this.props.location || JSON.stringify(newProps.location) !== JSON.stringify(this.props.location))) {
```
Comment (kamicut) [inline app/assets/scripts/components/map.js]: I don't know if JSON.stringify then match could become a performance problem later.

Comment (dereklieu) [inline app/assets/scripts/components/map.js]: Are you concerned we'll send more props, and that this function will fire more frequently, or are you concerned that the size of the input we'll feed to `JSON.stringify` will grow?

Comment (kamicut) [inline app/assets/scripts/components/map.js]: I'm flagging as potential for improving in case we think this could fire multiple times.

Comment (dereklieu) [inline app/assets/scripts/components/map.js]: It already fires multiple times, but we should be concerned if we think it'll fire many more times based on future code changes or data growth.

**Pull request merged**